pkgs/development/python-modules/distributed/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytest
+, pytest-repeat
+, pytest-faulthandler
+, pytest-timeout
+, mock
+, joblib
+, click
+, cloudpickle
+, dask
+, msgpack
+, psutil
+, six
+, sortedcontainers
+, tblib
+, toolz
+, tornado
+, zict
+, pyyaml
+, pythonOlder
+, futures
+, singledispatch
+}:
+
+buildPythonPackage rec {
+  pname = "distributed";
+  version = "1.22.1";
+
+  # get full repository need conftest.py to run tests
+  src = fetchFromGitHub {
+    owner = "dask";
+    repo = "${pname}";
+    rev = "${version}";
+    sha256 = "0xvx55rhbhlyys3kjndihwq6y6260qzy9mr3miclh5qddaiw2d5z";
+  };
+
+  checkInputs = [ pytest pytest-repeat pytest-faulthandler pytest-timeout mock joblib ];
+  propagatedBuildInputs = [
+      click cloudpickle dask msgpack psutil six
+      sortedcontainers tblib toolz tornado zict pyyaml
+  ] ++ lib.optional (pythonOlder "3.2") futures
+    ++ lib.optional (pythonOlder "3.4") singledispatch;
+
+  # tests take about 10-15 minutes
+  # ignore 5 cli tests out of 1000 total tests that fail due to subprocesses
+  # these tests are not critical to the library (only the cli)
+  checkPhase = ''
+    py.test distributed -m "not avoid-travis" -r s --timeout-method=thread --timeout=300 --durations=20 --ignore="distributed/cli/tests"
```

Comment:
Do you know if there's a way to set the timeout to infinity?

Reply:
Not sure. I have been following the test instructions that is used on travis.

https://github.com/dask/distributed/blob/master/continuous_integration/travis/run_tests.sh#L1

Reply:
From https://bitbucket.org/pytest-dev/pytest-timeout/:

> Setting a timeout to 0 seconds disables the timeout, so if you have a global timeout set you can still disable the timeout by using the mark.

However, I'd prefer to just get rid of `pytest-timeout` if that's possible. Otherwise, just set `--timeout=0`.

Reply:
Btw you should also try leaving out the other options if possible, especially `--durations` sounds like something we'd want to avoid. Hydra is usually under heavy load, so tests will run slower than usual.